src/commands/build.ts

```diff
@@ -236,11 +242,24 @@ export async function command(commandOptions: CommandOptions) {
         // We treat ".proxy.js" files special: we need to make sure that they exist on disk
         // in the final build, so we mark them to be written to disk at the next step.
         if (resolvedImportUrl.endsWith('.proxy.js')) {
-          allProxiedFiles.add(
-            resolvedImportUrl.startsWith('/')
-              ? path.resolve(cwd, spec)
-              : path.resolve(path.dirname(outLoc), spec),
-          );
+          // handle proxied files from web_modules
+          if (resolvedImportUrl.startsWith(config._webModulesPath)) {
+            allProxiedFiles.add(
+              path.resolve(
+                cwd,
+                config.devOptions.out,
+                config._webModulesPath.replace(/^[/\\]/, ''), // omit leading `/`
```

Comment:
nit: we should clean this value in config.ts so that it either always has.a leading slash or never does.